extension/parquet/parquet_extension.cpp

```diff
@@ -157,6 +159,89 @@ BindInfo ParquetGetBatchInfo(const FunctionData *bind_data) {
 	return bind_info;
 }
 
+static MultiFileReaderBindData BindSchema(ClientContext &context, vector<LogicalType> &return_types,
+                                          vector<string> &names, ParquetReadBindData &result, ParquetOptions &options) {
+	D_ASSERT(!options.schema.empty());
+	auto &file_options = options.file_options;
+	if (file_options.union_by_name || file_options.hive_partitioning) {
```

Comment:
Would be nice to throw a separate error for when hive auto detection is enabled. In the following case:
```
statement ok
COPY (
    SELECT 1 i1, 3 i3, 4 i4, 5 i5 UNION ALL
    SELECT 2 i1, 3 i3, 4 i4, 5 i5
) TO '__TEST_DIR__/partitioned' (FIELD_IDS {i1: 5, i3: 3, i4: 2, i5: 1}, PARTITION_BY i1, FORMAT parquet)

query IIII
SELECT *
FROM read_parquet('__TEST_DIR__/partitioned/*/*.parquet', schema=map {
                    1: {name: 'i1', type: 'BIGINT', default_value: NULL},
                    3: {name: 'i3', type: 'BIGINT', default_value: NULL},
                    4: {name: 'i4', type: 'BIGINT', default_value: 2},
                    5: {name: 'i5', type: 'BIGINT', default_value: NULL}
                  })
----
5	3	2	1
5	3	2	2
```

The binder error is thrown but hive_partitioning is not explicitly enabled.